src/Zenaton/v1/Client.php

```diff
@@ -342,26 +385,40 @@ public function scheduleWorkflow(WorkflowInterface $workflow, $cron)
      * @param string $workflowName Workflow class name
      * @param string $customId     Provided custom id
      *
+     * @throws ApiException if the API returns some errors
+     *
      * @return null|WorkflowInterface
      */
     public function findWorkflow($workflowName, $customId)
     {
-        $params = [
-            static::ATTR_ID => $customId,
-            static::ATTR_NAME => $workflowName,
-            static::ATTR_PROG => static::PROG,
-        ];
-
-        $response = $this->http->get($this->getInstanceWebsiteUrl($params));
-        if ($response->code === 404) {
-            return null;
-        }
+        $response = $this->graphqlClient->request(
+            Queries::WORKFLOW,
+            [
+                'customId' => $customId,
+                'environmentName' => $this->appEnv,
+                'programmingLanguage' => self::PROG,
+                'workflowName' => $workflowName,
+            ],
+            [
+                'App-Id' => $this->appId,
+                'Api-Token' => $this->apiToken,
+            ]
+        );
+
+        if (isset($response['errors'])) {
```

Comment:
It seems that a `try / catch` is necessary here since an `ApiException` will be thrown here.

Reply:
Actually, it's the opposite. I will remove the throwing of exception from the GraphQL client because an API call returning errors does not necessarily mean it should result in an exception. And that use case is precisely an illustration of that. The API will return a NOT_FOUND error, and yet we do not want to throw an exception.
Generating an exception from the client will make it harder to introspect the errors. So, as long as the API returns a result, even if it contains error, it's the caller of the graphql client that will take the responsibility of throwing an exception or not, depending on what the  user was doing and what are the errors.

Reply:
I updated the GraphQL client. It's no more returning an exception when the API is able to respond correctly, even if it returns errors.